homeassistant/components/brother/__init__.py

```diff
@@ -35,7 +35,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: BrotherConfigEntry) -> b
     await coordinator.async_config_entry_first_refresh()
 
     entry.runtime_data = coordinator
-    hass.data.setdefault(DOMAIN, {SNMP: snmp_engine})
+    hass.data[SNMP_ENGINE] = snmp_engine
```

Comment:
Just wondering, why not store a reference to the engine in `runtime_data`? It's already a singleton.

Ideally I think, we should only use `HassKey` for things that should be accessible from multiple different integrations.

Reply:
In theory, the SNMP engine can also be used by SNMP integration, but I have not checked this in practice. I will mark the PR as a draft and analyze it again. Thanks.

Reply:
Regarding this PR https://github.com/home-assistant/core/pull/117312, `runtime_data` will be removed when the configuration entry is unloaded. The SNMP engine can be shared between several config entries so if I understand correctly, it cannot be stored in `runtime_data`.

Reply:
> Regarding this PR #117312, `runtime_data` will be removed when the configuration entry is unloaded. The SNMP engine can be shared between several config entries so if I understand correctly, it cannot be stored in `runtime_data`. Am I missing something?

No, that sounds about right. Actually the `singleton` decorator already stores the engine in `hass.data` under `snmp_engine`. We should just reuse that key and we're set here.

https://github.com/home-assistant/core/blob/6f5e82009025e61a54339acf5601ad432d090cd8/homeassistant/components/brother/utils.py#L19-L20